azure-mgmt-resources/src/main/java/com/microsoft/azure/management/resources/fluentcore/arm/Region.java

```diff
@@ -156,6 +156,20 @@ public static Region fromName(String name) {
     }
 
     /**
+     * Parses a region label into a Region object and creates a new Region instance if not found among the existing ones.
+     * @param label a region label
+     * @return the parsed or created region
+     */
+    public static Region fromLabel(String label) {
+        Region region = VALUES_BY_LABEL.get(label.toLowerCase());
+        if (region != null) {
+            return region;
+        } else {
+            return Region.create(label, label);
+        }
+    }
+
```

Comment:
As per Azure REST API guideline related to region - when server validates the region name it ignores the spaces and casing. This means "SouthEast Asia", "South East Asia", "southeastasia", all these are valid and represents the same region. Hence if a service returns region text in any of the above format, I am not sure we can say its a server bug based on api spec.

In our region enum we introduced the property `Label` because thats the display name used by azure portal. Afaik, in api spec there is nothing like label, there is only name.


Feel like we should remove the label property from the region enum. Whatever region name server returns we could remove spaces and lower case it (by following spec).

`Label` is used as key in `fromLabel`, this can cause duplicate entries in the map

southeast asia -> southeast asia
south east asia -> south east asia
southeastasia -> southeastasia
...

After all, only purpose of label seems, If consumer of SDK want to use it as display name in their UI app, but usually display names goes to resource file and there will be different resource files depends on the languages (Chinese, Spanish etc..) the app supports.

Btw [Region enum](https://github.com/Azure/azure-sdk-for-net/blob/Fluent/src/ResourceManagement/ResourceManager/Microsoft.Azure.Management.ResourceManager.Fluent/Region.cs) in .NET fluent does not have `Label`.

@martinsawicki pls share your thoughts?

@anudeepsharma we had a discussion once about this, can you please confirm or correct my understanding of server side handling of region, how DTLab parse this?

\\cc @jianghaolu